internal/server/middleware/grpc/middleware.go

```diff
@@ -234,16 +234,29 @@ func CacheUnaryInterceptor(cache cache.Cacher, logger *zap.Logger) grpc.UnarySer
 	}
 }
 
+type namespaceKeyer interface {
+	GetNamespaceKey() string
+}
+
 type flagKeyer interface {
+	namespaceKeyer
 	GetKey() string
 }
 
 type variantFlagKeyger interface {
+	namespaceKeyer
 	GetFlagKey() string
 }
 
-func flagCacheKey(key string) string {
-	k := fmt.Sprintf("f:%s", key)
+func flagCacheKey(namespaceKey, key string) string {
+	var k string
+	// for backwards compatibility
+	if namespaceKey != "" {
+		k = fmt.Sprintf("f:%s:%s", namespaceKey, key)
+	} else {
+		k = fmt.Sprintf("f:%s", key)
+	}
```

Comment:
The backwards compatibility here refers to other dbs as a data source other than sqlite correct? Or is there more it is referring to?